importers/src/osis-sword-module/tests/OsisParser.test.ts

```diff
@@ -1,61 +1,53 @@
+        const psalm23verse2 = [psalmsXML.verses[1]];
+        const bookJson = getBibleEngineInputFromXML([{ intro: '', verses: psalm23verse2 }]);
+        const footnoteText = 'waters of rest';
+        expect(JSON.stringify(bookJson)).toEqual(expect.stringContaining(footnoteText));
+    });
 
-  test('section header is included in psalms', () => {
-    const bookJson = getBibleEngineInputFromXML([
-      { intro: '', verses: psalmsXML.verses }
-    ]);
-    const sectionHeader = 'The LORD Is My Shepherd';
-    expect(JSON.stringify(bookJson)).toEqual(
-      expect.stringContaining(sectionHeader)
-    )
-  })
+    test('section header is included in psalms', () => {
+        const bookJson = getBibleEngineInputFromXML([{ intro: '', verses: psalmsXML.verses }]);
+        const sectionHeader = 'The LORD Is My Shepherd';
+        expect(JSON.stringify(bookJson)).toEqual(expect.stringContaining(sectionHeader));
+    });
 
-  test('psalm title is included', () => {
-    const bookJson = getBibleEngineInputFromXML([
-      { intro: '', verses: psalmsXML.verses }
-    ]);
-    let titleGroupExists = false;
-    const queue = [bookJson[0]];
-    while (queue.length) {
-      const top: any = queue.pop();
-      if (top['groupType'] === 'title') {
-        titleGroupExists = true;
-        return;
-      }
-      if ('contents' in top) {
-        queue.push(...top['contents'])
-      }
-    }
-    const partOfPsalmTitle = 'David'
-    expect(titleGroupExists).toBe(true)
-    expect(JSON.stringify(bookJson)).toEqual(
-      expect.stringContaining(partOfPsalmTitle)
-    );
-  })
+    test('psalm title is included', () => {
+        const bookJson = getBibleEngineInputFromXML([{ intro: '', verses: psalmsXML.verses }]);
+        let titleGroupExists = false;
+        const queue = [bookJson[0]];
+        while (queue.length) {
+            const top: any = queue.pop();
+            if (top['groupType'] === 'title') {
+                titleGroupExists = true;
+                return;
+            }
+            if ('contents' in top) {
+                queue.push(...top['contents']);
+            }
+        }
+        const partOfPsalmTitle = 'David';
+        expect(titleGroupExists).toBe(true);
+        expect(JSON.stringify(bookJson)).toEqual(expect.stringContaining(partOfPsalmTitle));
+    });
 
-  describe('Chinese Union Version Simplified with Strongs', () => {
-    const genesisXml: types.ChapterXML = require('./Gen1CuvXmlResult.json');
-    test('nonempty content returned', () => {
-      const bookJson = getBibleEngineInputFromXML([
-        { intro: '', verses: genesisXml.verses }
-      ]);
-      console.log(JSON.stringify(bookJson[0], null, 2))
-      expect(bookJson.length).toBeGreaterThan(0)
-    })
-  })
+    test('Erroneous Psalm 42 headings arent included in Psalm 41 output', () => {
+        const bookJson = getBibleEngineInputFromXML([{ intro: '', verses: [Psalm41] }]);
```

Comment:
i'm not sure if this test really tests anything since we know that the test data you are including here contains the bug - or am i missing something?

Reply:
agree. I removed this test